benchmark/utils/hip_linops.hip.hpp

```diff
@@ -315,6 +320,191 @@ class HipspHybrid
 };
 
 
+namespace hip {
+namespace ilu {
+
+
+class NamedLambdaOperation : public gko::Operation {
+public:
+    NamedLambdaOperation(
+        std::string name,
+        std::function<void(std::shared_ptr<const gko::HipExecutor>)> op)
+        : name_{std::move(name)}, op_{std::move(op)}
+    {}
+    void run(std::shared_ptr<const gko::HipExecutor> exec) const override
+    {
+        op_(exec);
+    }
+    const char *get_name() const noexcept override { return name_.c_str(); }
+
+private:
+    std::string name_;
+    std::function<void(std::shared_ptr<const gko::HipExecutor>)> op_;
+};
+
+
+GKO_REGISTER_OPERATION(should_transpose, lower_trs::should_perform_transpose);
+GKO_REGISTER_OPERATION(init_struct_l, lower_trs::init_struct);
+GKO_REGISTER_OPERATION(init_struct_u, upper_trs::init_struct);
+GKO_REGISTER_OPERATION(generate_l, lower_trs::generate);
+GKO_REGISTER_OPERATION(generate_u, upper_trs::generate);
+GKO_REGISTER_OPERATION(solve_l, lower_trs::solve);
+GKO_REGISTER_OPERATION(solve_u, upper_trs::solve);
+
+
+}  // namespace ilu
+}  // namespace hip
+
+
+template <typename ValueType = gko::default_precision,
+          typename IndexType = gko::int32>
+class HipspIlu
```

Comment:
Should this be moved to Ginkgo itself?

Reply:
Do you mean, both CuspspIlu and HIPspilu ? That would be nice. I guess then you have something similar to the strategies in CSR to select between the implementations ?